rust/datafusion/src/logical_plan/expr.rs

```diff
@@ -237,6 +245,25 @@ impl Expr {
             Expr::Wildcard => Err(DataFusionError::Internal(
                 "Wildcard expressions are not valid in a logical query plan".to_owned(),
             )),
+            Expr::Extract {
+                date_part: DatePart::Hour,
+                expr,
+            } => {
+                let inner = expr.get_type(schema)?;
+                match inner {
+                    DataType::Date32(_)
+                    | DataType::Date64(_)
+                    | DataType::Time32(_)
+                    | DataType::Time64(_)
+                    | DataType::Timestamp(arrow::datatypes::TimeUnit::Nanosecond, None) => {
```

Comment:
Is there a particular reason to only support `Timestamp(Nanosecond)`? `Millisecond,Second,Microsecond` should also work, no?

Reply:
I wasn't convinced yet the `hour` kernel was working correctly, and it's currently a bit hard to test from DataFusion. But I am convinced now that it should work as expected. I think it can use some more tests at the Arrow side as well.